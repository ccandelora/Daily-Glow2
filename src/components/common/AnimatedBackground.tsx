import React, { useEffect, useRef } from 'react';
import { Animated, StyleSheet, View, Dimensions } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import theme from '@/constants/theme';

interface AnimatedBackgroundProps {
  intensity?: 'light' | 'medium' | 'strong';
  animated?: boolean;
}

export const AnimatedBackground: React.FC<AnimatedBackgroundProps> = ({ 
  intensity = 'medium',
  animated = true 
}) => {
  const fadeAnim = useRef(new Animated.Value(0)).current;
  const rotateAnim = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (animated) {
      // Fade in
      Animated.timing(fadeAnim, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: true,
      }).start();

      // Faster rotation
      const rotateAnimation = Animated.loop(
        Animated.sequence([
          Animated.timing(rotateAnim, {
            toValue: 1,
            duration: 30000, // 30 seconds per rotation
            useNativeDriver: true,
          }),
          Animated.timing(rotateAnim, {
            toValue: 0,
            duration: 0,
            useNativeDriver: true,
          })
        ])
      );

      rotateAnimation.start();
    } else {
      fadeAnim.setValue(1);
    }
  }, [animated]);

  const spin = rotateAnim.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg']
  });

  return (
    <View style={styles.container}>
      {/* Static base gradient */}
      <LinearGradient
        colors={[
          'rgba(255,255,255,0.92)',
          'rgba(235,230,255,0.95)',
          'rgba(255,255,255,0.92)',
        ]}
        style={styles.baseGradient}
        start={{ x: 0, y: 0 }}
        end={{ x: 0, y: 1 }}
      />

      <Animated.View
        style={[
          styles.gradientContainer,
          {
            opacity: fadeAnim,
            transform: [{ rotate: spin }]
          }
        ]}
      >
        {/* Brighter purple glow */}
        <LinearGradient
          colors={[
            'transparent',
            `${theme.COLORS.primary.blue}15`,
            `${theme.COLORS.primary.blue}20`,
            `${theme.COLORS.primary.blue}15`,
            'transparent',
          ]}
          style={styles.gradient}
          start={{ x: 0.3, y: 0.3 }}
          end={{ x: 0.7, y: 0.7 }}
          locations={[0, 0.3, 0.5, 0.7, 1]}
        />

        {/* Stronger accent highlights */}
        <LinearGradient
          colors={[
            'transparent',
            'rgba(230,230,255,0.35)',
            'transparent'
          ]}
          style={styles.gradient}
          start={{ x: 0, y: 0.5 }}
          end={{ x: 1, y: 0.5 }}
        />

        {/* Enhanced color variation */}
        <LinearGradient
          colors={[
            'transparent',
            `${theme.COLORS.primary.blue}12`,
            'transparent'
          ]}
          style={styles.gradient}
          start={{ x: 0.5, y: 0 }}
          end={{ x: 0.5, y: 1 }}
        />
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: 'white',
  },
  baseGradient: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  gradientContainer: {
    position: 'absolute',
    top: -400,
    left: -400,
    right: -400,
    bottom: -400,
  },
  gradient: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    transform: [{ scale: 2 }],
  },
}); 