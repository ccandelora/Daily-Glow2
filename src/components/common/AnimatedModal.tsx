import React, { useEffect } from 'react';
import {
  View,
  Modal,
  StyleSheet,
  TouchableOpacity,
  Animated,
  Dimensions,
  ViewStyle,
} from 'react-native';
import theme from '../../constants/theme';
import { Typography } from './Typography';

interface AnimatedModalProps {
  visible: boolean;
  onClose: () => void;
  title?: string;
  children: React.ReactNode;
  style?: ViewStyle;
}

export const AnimatedModal: React.FC<AnimatedModalProps> = ({
  visible,
  onClose,
  title,
  children,
  style,
}) => {
  const slideAnim = React.useRef(new Animated.Value(0)).current;
  const fadeAnim = React.useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (visible) {
      Animated.parallel([
        Animated.timing(slideAnim, {
          toValue: 1,
          duration: 300,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 1,
          duration: 300,
          useNativeDriver: true,
        }),
      ]).start();
    } else {
      Animated.parallel([
        Animated.timing(slideAnim, {
          toValue: 0,
          duration: 200,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 0,
          duration: 200,
          useNativeDriver: true,
        }),
      ]).start();
    }
  }, [visible, slideAnim, fadeAnim]);

  const { height } = Dimensions.get('window');

  return (
    <Modal
      transparent
      visible={visible}
      onRequestClose={onClose}
      animationType="none"
    >
      <Animated.View
        style={[
          styles.overlay,
          {
            opacity: fadeAnim,
          },
        ]}
      >
        <TouchableOpacity
          style={styles.backdrop}
          onPress={onClose}
          activeOpacity={1}
        />
        <Animated.View
          style={[
            styles.content,
            {
              transform: [
                {
                  translateY: slideAnim.interpolate({
                    inputRange: [0, 1],
                    outputRange: [height, 0],
                  }),
                },
              ],
            },
            style,
          ]}
        >
          {title && (
            <View style={styles.header}>
              <Typography variant="h3">{title}</Typography>
              <TouchableOpacity onPress={onClose} style={styles.closeButton}>
                <Typography variant="h3">Ã—</Typography>
              </TouchableOpacity>
            </View>
          )}
          {children}
        </Animated.View>
      </Animated.View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  overlay: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    justifyContent: 'flex-end',
  },
  backdrop: {
    ...StyleSheet.absoluteFillObject,
  },
  content: {
    backgroundColor: theme.COLORS.ui.background,
    borderTopLeftRadius: theme.BORDER_RADIUS.xl,
    borderTopRightRadius: theme.BORDER_RADIUS.xl,
    padding: theme.SPACING.lg,
    maxHeight: '90%',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: theme.SPACING.md,
  },
  closeButton: {
    padding: theme.SPACING.sm,
    margin: -theme.SPACING.sm,
  },
}); 