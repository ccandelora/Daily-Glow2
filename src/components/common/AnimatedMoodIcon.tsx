import React, { useEffect } from 'react';
import { Animated, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface AnimatedMoodIconProps {
  size?: number;
  color: string;
  style?: ViewStyle;
  children: React.ReactNode;
  active?: boolean;
}

export const AnimatedMoodIcon: React.FC<AnimatedMoodIconProps> = ({
  size = 60,
  color,
  style,
  children,
  active = false,
}) => {
  const scaleAnim = React.useRef(new Animated.Value(1)).current;
  const opacityAnim = React.useRef(new Animated.Value(0.6)).current;

  useEffect(() => {
    Animated.parallel([
      Animated.spring(scaleAnim, {
        toValue: active ? 1.1 : 1,
        useNativeDriver: true,
        damping: 10,
      }),
      Animated.timing(opacityAnim, {
        toValue: active ? 1 : 0.6,
        duration: 200,
        useNativeDriver: true,
      }),
    ]).start();
  }, [active, scaleAnim, opacityAnim]);

  return (
    <Animated.View
      style={[
        styles.container,
        {
          width: size,
          height: size,
          backgroundColor: color,
          transform: [{ scale: scaleAnim }],
          opacity: opacityAnim,
        },
        style,
      ]}
    >
      {children}
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: theme.BORDER_RADIUS.circle,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
  },
}); 