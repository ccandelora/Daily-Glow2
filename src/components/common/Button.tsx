import React from 'react';
import { TouchableOpacity, StyleSheet, ViewStyle, TextStyle, ActivityIndicator, StyleProp } from 'react-native';
import { Typography } from './Typography';
import theme from '@/constants/theme';

interface ButtonProps {
  title: string;
  onPress: () => void;
  variant?: 'primary' | 'secondary' | 'outline';
  size?: 'small' | 'medium' | 'large';
  disabled?: boolean;
  loading?: boolean;
  style?: ViewStyle;
  textStyle?: TextStyle;
}

export const Button: React.FC<ButtonProps> = ({
  title,
  onPress,
  variant = 'primary',
  size = 'medium',
  disabled = false,
  loading = false,
  style,
  textStyle,
}) => {
  return (
    <TouchableOpacity
      style={[
        styles.base,
        styles[variant],
        styles[size],
        disabled && styles.disabled,
        style,
      ]}
      onPress={onPress}
      disabled={disabled || loading}
    >
      {loading ? (
        <ActivityIndicator
          color={variant === 'outline' ? theme.COLORS.primary.green : theme.COLORS.ui.background}
          size="small"
        />
      ) : (
        <Typography
          variant="button"
          style={[
            styles.text,
            styles[`${variant}Text` as keyof typeof styles] as TextStyle,
            disabled && styles.disabledText,
            textStyle,
          ].filter(Boolean) as TextStyle}
        >
          {title}
        </Typography>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  base: {
    borderRadius: theme.BORDER_RADIUS.lg,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    shadowColor: theme.COLORS.ui.accent,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 8,
    elevation: 4,
  },
  primary: {
    backgroundColor: theme.COLORS.ui.accent,
    borderWidth: 0,
    shadowColor: theme.COLORS.ui.accent,
    shadowOpacity: 0.5,
    shadowRadius: 12,
  },
  secondary: {
    backgroundColor: 'rgba(65, 105, 225, 0.1)',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.accent,
  },
  outline: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.border,
  },
  small: {
    paddingVertical: theme.SPACING.sm,
    paddingHorizontal: theme.SPACING.lg,
    minWidth: 100,
  },
  medium: {
    paddingVertical: theme.SPACING.md,
    paddingHorizontal: theme.SPACING.xl,
    minWidth: 150,
  },
  large: {
    paddingVertical: theme.SPACING.lg,
    paddingHorizontal: theme.SPACING.xl * 1.5,
    minWidth: 200,
  },
  disabled: {
    backgroundColor: theme.COLORS.ui.disabled,
    borderColor: 'transparent',
    shadowOpacity: 0,
  },
  text: {
    textAlign: 'center',
  },
  primaryText: {
    color: theme.COLORS.ui.text,
    textShadowColor: 'rgba(0, 0, 0, 0.2)',
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 2,
  },
  secondaryText: {
    color: theme.COLORS.ui.accent,
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 0 },
    textShadowRadius: 4,
  },
  outlineText: {
    color: theme.COLORS.ui.text,
  },
  disabledText: {
    color: theme.COLORS.ui.textSecondary,
  },
}); 