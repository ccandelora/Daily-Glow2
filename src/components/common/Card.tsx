import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import theme from '../../constants/theme';

interface CardProps {
  children: React.ReactNode;
  style?: ViewStyle;
  variant?: 'default' | 'elevated';
}

export const Card: React.FC<CardProps> = ({
  children,
  style,
  variant = 'default',
}) => {
  return (
    <View
      style={[
        styles.card,
        variant === 'elevated' && styles.elevated,
        style,
      ]}
    >
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: theme.COLORS.ui.card,
    borderRadius: theme.BORDER_RADIUS.lg,
    padding: theme.SPACING.md,
    marginVertical: theme.SPACING.sm,
  },
  elevated: {
    shadowColor: theme.COLORS.ui.text,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
}); 