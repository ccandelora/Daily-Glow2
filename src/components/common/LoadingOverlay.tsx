import React from 'react';
import { View, StyleSheet, ActivityIndicator } from 'react-native';
import { Typography } from './Typography';
import theme from '@/constants/theme';

interface LoadingOverlayProps {
  message?: string;
}

export const LoadingOverlay: React.FC<LoadingOverlayProps> = ({ message }) => {
  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <ActivityIndicator size="large" color={theme.COLORS.primary.green} />
        {message && (
          <Typography
            variant="body"
            color={theme.COLORS.ui.textSecondary}
            style={styles.message}
          >
            {message}
          </Typography>
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: 'rgba(0, 0, 0, 0.7)',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 1000,
  },
  content: {
    backgroundColor: theme.COLORS.ui.card,
    padding: theme.SPACING.xl,
    borderRadius: theme.BORDER_RADIUS.lg,
    alignItems: 'center',
  },
  message: {
    marginTop: theme.SPACING.md,
    textAlign: 'center',
  },
}); 