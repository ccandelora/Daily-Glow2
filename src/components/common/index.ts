export * from './Button';
export * from './Card';
export * from './Typography';
export * from './Input';
export * from './ProgressBar';
export * from './Header';
export * from './AnimatedModal';
export * from './AnimatedMoodIcon';
export * from './EmotionWheel';
export { LoadingSpinner } from './LoadingSpinner';
export { Toast } from './Toast';
export { LoadingOverlay } from './LoadingOverlay';
export { DailyChallenge } from './DailyChallenge';
export { NotificationBadge } from './NotificationBadge';
export { TabBar } from './TabBar'; 