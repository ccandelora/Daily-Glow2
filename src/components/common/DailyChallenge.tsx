import React, { useState, useEffect } from 'react';
import { View, StyleSheet, TouchableOpacity, Animated, ViewStyle, StyleProp } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Typography } from './Typography';
import { Card } from './Card';
import { Button } from './Button';
import { Input } from './Input';
import { useChallenges } from '@/contexts/ChallengesContext';
import { useAppState } from '@/contexts/AppStateContext';
import theme from '@/constants/theme';
import { Ionicons } from '@expo/vector-icons';
import { useRouter } from 'expo-router';

interface DailyChallengeProps {
  onComplete?: () => void;
  style?: StyleProp<ViewStyle>;
}

const typeIcons = {
  mood: 'heart-outline',
  gratitude: 'happy-outline',
  mindfulness: 'leaf-outline',
  creative: 'brush-outline',
} as const;

const getPromptForType = (type: string) => {
  switch (type) {
    case 'mood':
      return 'Describe your current mood and feelings...';
    case 'gratitude':
      return 'Share what you\'re grateful for today...';
    case 'mindfulness':
      return 'Describe your mindfulness experience...';
    case 'creative':
      return 'Share your creative response...';
    default:
      return 'Share your thoughts...';
  }
};

const getMinLengthForType = (type: string) => {
  return type === 'creative' ? 20 : 10;
};

export const DailyChallenge: React.FC<DailyChallengeProps> = ({ onComplete, style }) => {
  const { dailyChallenge, userStats, completeChallenge, refreshDailyChallenge, userChallenges } = useChallenges();
  const { showError } = useAppState();
  const [completionText, setCompletionText] = useState('');
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [localPoints, setLocalPoints] = useState(0);
  const [completedToday, setCompletedToday] = useState(0);
  const [showLimitMessage, setShowLimitMessage] = useState(false);
  const router = useRouter();

  useEffect(() => {
    console.log('DailyChallenge state:', {
      hasDailyChallenge: !!dailyChallenge,
      dailyChallengeId: dailyChallenge?.id,
      completedToday,
      showLimitMessage,
      userChallengesCount: userChallenges.length
    });
  }, []);

  useEffect(() => {
    if (userStats?.total_points !== undefined) {
      setLocalPoints(userStats.total_points);
      
      // Count completed challenges for today using local timezone
      const today = new Date();
      today.setHours(0, 0, 0, 0); // Set to start of day in local timezone
      
      const completedCount = userChallenges.filter(challenge => {
        if (!challenge.completed_at) return false;
        
        // Convert UTC timestamp to local date for comparison
        const completionDate = new Date(challenge.completed_at);
        const localCompletionDate = new Date(
          completionDate.getFullYear(),
          completionDate.getMonth(),
          completionDate.getDate()
        );
        
        const isCompleted = challenge.status === 'completed';
        const isToday = localCompletionDate.getTime() === today.getTime();
        
        return isCompleted && isToday;
      }).length;
      
      setCompletedToday(completedCount);
      setShowLimitMessage(completedCount >= 2);
    }
  }, [userStats, userChallenges]);

  // Single useEffect for refresh logic
  useEffect(() => {
    const setupMidnightRefresh = () => {
      const now = new Date();
      const tomorrow = new Date(now);
      tomorrow.setDate(tomorrow.getDate() + 1);
      tomorrow.setHours(0, 0, 0, 0);
      
      const timeUntilMidnight = tomorrow.getTime() - now.getTime();
      
      // Only refresh if we don't have a challenge
      if (!dailyChallenge) {
        refreshDailyChallenge();
      }
      
      return setTimeout(() => {
        refreshDailyChallenge();
        setupMidnightRefresh();
      }, timeUntilMidnight);
    };

    const timer = setupMidnightRefresh();
    return () => clearTimeout(timer);
  }, [dailyChallenge]); // Only re-run if dailyChallenge changes

  if (showLimitMessage) {
    return (
      <View style={styles.glowContainer}>
        <Card style={styles.container}>
          <View style={styles.completionContent}>
            <Ionicons 
              name="checkmark-circle-outline" 
              size={48} 
              color={theme.COLORS.primary.green}
              style={styles.completionIcon}
            />
            <Typography variant="h2" style={styles.completionTitle}>
              Daily Challenges Complete!
            </Typography>
            <Typography style={styles.completionMessage}>
              Great job! You've completed your daily challenges. Come back tomorrow for new challenges and more opportunities to earn points.
            </Typography>
            <View style={styles.completionPoints}>
              <Typography variant="h1" style={styles.completionPointsValue}>
                {localPoints}
              </Typography>
              <Typography style={styles.completionPointsLabel}>
                Total Points
              </Typography>
            </View>
            <Button
              title="View Your Achievements"
              onPress={() => router.push('/(app)/achievements')}
              variant="secondary"
              style={styles.completionButton}
            />
          </View>
        </Card>
      </View>
    );
  }

  if (!dailyChallenge) {
    return (
      <View style={styles.glowContainer}>
        <Card style={styles.container}>
          <Typography variant="body" style={styles.limitMessage}>
            No challenges available right now. Please check back later.
          </Typography>
        </Card>
      </View>
    );
  }

  const handleComplete = async () => {
    if (completedToday >= 2) {
      setShowLimitMessage(true);
      return;
    }

    try {
      setIsSubmitting(true);
      
      const minLength = getMinLengthForType(dailyChallenge.type);
      if (!completionText.trim() || completionText.length < minLength) {
        showError(`Please provide a response with at least ${minLength} characters`);
        return;
      }

      await completeChallenge(dailyChallenge.id, completionText.trim());
      await refreshDailyChallenge();
      setCompletionText('');
      onComplete?.();
    } catch (error) {
      if (error instanceof Error && error.message.includes('Daily challenge limit reached')) {
        setShowLimitMessage(true);
      } else {
        showError(error instanceof Error ? error.message : 'Failed to complete challenge');
      }
    } finally {
      setIsSubmitting(false);
    }
  };

  const currentStreak = userStats?.current_streak ?? 0;
  const level = userStats?.level ?? 1;

  return (
    <View style={styles.glowContainer}>
      <Card style={[styles.container, style]}>
        <View style={styles.header}>
          <View style={styles.titleContainer}>
            <Ionicons 
              name={typeIcons[dailyChallenge.type as keyof typeof typeIcons]} 
              size={24} 
              color={theme.COLORS.primary.green}
            />
            <Typography variant="h3" style={styles.title}>
              Daily Challenge
            </Typography>
          </View>
          <View style={styles.challengeStats}>
            <Typography variant="caption" color={theme.COLORS.ui.textSecondary}>
              Level {level}
            </Typography>
            <Typography 
              variant="h3" 
              color={theme.COLORS.primary.green}
              style={styles.pointsText}
            >
              {localPoints} pts
            </Typography>
          </View>
        </View>

        <Typography variant="h3" style={styles.challengeTitle}>
          {dailyChallenge.title}
        </Typography>
        
        <Typography 
          variant="body" 
          style={{ color: theme.COLORS.ui.textSecondary }}
        >
          {dailyChallenge.description}
        </Typography>

        <Input
          multiline
          value={completionText}
          onChangeText={setCompletionText}
          placeholder={getPromptForType(dailyChallenge.type)}
          style={styles.input}
          maxLength={500}
        />

        <View style={styles.footer}>
          <View style={styles.pointsContainer}>
            <Ionicons 
              name="star-outline" 
              size={16} 
              color={theme.COLORS.primary.green}
            />
            <Typography 
              variant="body"
              style={styles.points}
              color={theme.COLORS.primary.green}
            >
              {dailyChallenge.points} points
            </Typography>
          </View>
          <Button
            title="Complete Challenge"
            onPress={handleComplete}
            style={styles.button}
            disabled={isSubmitting}
          />
        </View>

        {currentStreak > 0 && (
          <View style={styles.streakContainer}>
            <Ionicons 
              name="flame-outline" 
              size={16} 
              color={theme.COLORS.primary.red}
            />
            <Typography 
              variant="caption" 
              style={{ color: theme.COLORS.primary.red }}
            >
              {currentStreak} day streak!
            </Typography>
          </View>
        )}
      </Card>
    </View>
  );
};

const styles = StyleSheet.create({
  glowContainer: {
    width: '100%',
    alignItems: 'center',
    marginVertical: theme.SPACING.lg,
    paddingHorizontal: theme.SPACING.lg,
  },
  container: {
    width: '100%',
    padding: theme.SPACING.xl,
    backgroundColor: 'rgba(255, 255, 255, 0.95)',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginBottom: theme.SPACING.md,
    width: '100%',
    flexWrap: 'wrap',
    gap: theme.SPACING.md,
  },
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: theme.SPACING.sm,
    flex: 1,
    minWidth: 150,
  },
  title: {
    marginLeft: theme.SPACING.xs,
    flexShrink: 1,
  },
  challengeStats: {
    alignItems: 'flex-end',
    minWidth: 100,
    flexShrink: 0,
  },
  challengeTitle: {
    marginBottom: theme.SPACING.sm,
  },
  description: {
    marginBottom: theme.SPACING.lg,
  },
  input: {
    height: 120,
    marginBottom: theme.SPACING.lg,
    textAlignVertical: 'top',
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  pointsContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    gap: theme.SPACING.xs,
  },
  points: {
    fontWeight: theme.FONTS.weights.semibold,
    fontSize: theme.FONTS.sizes.md,
  },
  button: {
    flex: 1,
    marginLeft: theme.SPACING.md,
  },
  streakContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: theme.SPACING.md,
    paddingTop: theme.SPACING.md,
    borderTopWidth: 1,
    borderTopColor: theme.COLORS.ui.border,
  },
  pointsText: {
    fontWeight: theme.FONTS.weights.bold,
    color: theme.COLORS.primary.green,
    marginTop: theme.SPACING.xs,
    textAlign: 'right',
    minWidth: 100,
    flexWrap: 'wrap',
    flexShrink: 0,
  },
  limitContainer: {
    padding: theme.SPACING.lg,
    alignItems: 'center',
    width: '100%',
  },
  limitHeader: {
    alignItems: 'center',
    marginBottom: theme.SPACING.xl,
    width: '100%',
  },
  limitTitle: {
    textAlign: 'center',
    color: theme.COLORS.primary.green,
    marginTop: theme.SPACING.md,
    fontSize: theme.FONTS.sizes.xxl,
    paddingHorizontal: theme.SPACING.md,
    width: '100%',
    flexWrap: 'wrap',
  },
  limitMessage: {
    textAlign: 'center',
    color: theme.COLORS.ui.textSecondary,
    marginBottom: theme.SPACING.xl,
    lineHeight: 24,
    paddingHorizontal: theme.SPACING.lg,
    width: '100%',
  },
  statsContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    marginBottom: theme.SPACING.xl,
    paddingHorizontal: theme.SPACING.lg,
  },
  pointsDisplay: {
    alignItems: 'center',
    flex: 1,
    paddingHorizontal: theme.SPACING.lg,
    width: '100%',
    maxWidth: 200,
  },
  pointsValue: {
    color: theme.COLORS.primary.green,
    fontSize: theme.FONTS.sizes.xxxl,
    fontWeight: theme.FONTS.weights.bold,
    textAlign: 'center',
    width: '100%',
    flexShrink: 0,
    minWidth: 150,
    paddingHorizontal: theme.SPACING.md,
  },
  pointsLabel: {
    color: theme.COLORS.ui.textSecondary,
    marginTop: theme.SPACING.xs,
    textAlign: 'center',
    width: '100%',
  },
  streakDisplay: {
    alignItems: 'center',
    flex: 1,
    borderLeftWidth: 1,
    borderLeftColor: theme.COLORS.ui.border,
    paddingLeft: theme.SPACING.lg,
  },
  streakValue: {
    color: theme.COLORS.primary.red,
    marginTop: theme.SPACING.xs,
    fontSize: theme.FONTS.sizes.xxl,
    fontWeight: theme.FONTS.weights.bold,
  },
  streakLabel: {
    color: theme.COLORS.ui.textSecondary,
  },
  achievementsButton: {
    marginTop: 0,
    paddingVertical: theme.SPACING.md,
    width: '100%',
  },
  completionContainer: {
    width: '100%',
    backgroundColor: 'rgba(255, 255, 255, 0.95)',
    borderRadius: theme.BORDER_RADIUS.lg,
  },
  completionContent: {
    padding: theme.SPACING.xl,
    alignItems: 'center',
    paddingVertical: theme.SPACING.xl * 1.5,
  },
  completionIcon: {
    marginBottom: theme.SPACING.lg,
  },
  completionTitle: {
    color: theme.COLORS.primary.green,
    fontSize: 28,
    fontWeight: theme.FONTS.weights.semibold,
    textAlign: 'center',
    marginBottom: theme.SPACING.md,
  },
  completionMessage: {
    color: theme.COLORS.ui.textSecondary,
    textAlign: 'center',
    marginBottom: theme.SPACING.xl * 1.5,
    lineHeight: 24,
    fontSize: theme.FONTS.sizes.md,
  },
  completionPoints: {
    alignItems: 'center',
    marginBottom: theme.SPACING.xl * 1.5,
  },
  completionPointsValue: {
    color: theme.COLORS.primary.green,
    fontSize: 72,
    fontWeight: theme.FONTS.weights.bold,
    textAlign: 'center',
    marginBottom: theme.SPACING.xs,
    lineHeight: 80,
  },
  completionPointsLabel: {
    color: theme.COLORS.ui.textSecondary,
    textAlign: 'center',
    fontSize: theme.FONTS.sizes.lg,
  },
  completionButton: {
    width: '100%',
    paddingVertical: theme.SPACING.md,
    marginTop: theme.SPACING.md,
  },
}); 