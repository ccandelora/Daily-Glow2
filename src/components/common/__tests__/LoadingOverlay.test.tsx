import React from 'react';
import { render } from '@testing-library/react-native';
import { LoadingOverlay } from '@/components/common/LoadingOverlay';
import { ActivityIndicator } from 'react-native';

describe('LoadingOverlay Component', () => {
  it('renders correctly without a message', () => {
    const { toJSON, UNSAFE_getAllByType, queryByText } = render(
      <LoadingOverlay />
    );
    
    // Should have an ActivityIndicator
    const activityIndicators = UNSAFE_getAllByType(ActivityIndicator);
    expect(activityIndicators.length).toBe(1);
    
    // Should not have any text message
    expect(queryByText('Loading...')).toBeFalsy();
    
    // Should match snapshot
    expect(toJSON()).toMatchSnapshot();
  });
  
  it('renders correctly with a message', () => {
    const testMessage = 'Loading your data...';
    const { toJSON, getByText, UNSAFE_getAllByType } = render(
      <LoadingOverlay message={testMessage} />
    );
    
    // Should have an ActivityIndicator
    const activityIndicators = UNSAFE_getAllByType(ActivityIndicator);
    expect(activityIndicators.length).toBe(1);
    
    // Should display the correct message
    expect(getByText(testMessage)).toBeTruthy();
    
    // Should match snapshot
    expect(toJSON()).toMatchSnapshot();
  });
  
  it('applies the correct styling', () => {
    const { toJSON } = render(<LoadingOverlay />);
    const tree = toJSON();
    
    // Check that the container has the expected styles
    expect(tree).toHaveProperty('props.style');
    
    // Should match snapshot
    expect(tree).toMatchSnapshot();
  });
}); 