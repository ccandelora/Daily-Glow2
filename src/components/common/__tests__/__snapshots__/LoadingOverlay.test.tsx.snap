// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay Component applies the correct styling 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.7)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(38, 20, 60, 0.85)",
        "borderRadius": 16,
        "padding": 32,
      }
    }
  >
    <ActivityIndicator
      color="#00C853"
      size="large"
    />
  </View>
</View>
`;

exports[`LoadingOverlay Component renders correctly with a message 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.7)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(38, 20, 60, 0.85)",
        "borderRadius": 16,
        "padding": 32,
      }
    }
  >
    <ActivityIndicator
      color="#00C853"
      size="large"
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 16,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          {
            "color": "rgba(255, 255, 255, 0.7)",
          },
          null,
          {
            "marginTop": 16,
            "textAlign": "center",
          },
        ]
      }
    >
      Loading your data...
    </Text>
  </View>
</View>
`;

exports[`LoadingOverlay Component renders correctly without a message 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(0, 0, 0, 0.7)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(38, 20, 60, 0.85)",
        "borderRadius": 16,
        "padding": 32,
      }
    }
  >
    <ActivityIndicator
      color="#00C853"
      size="large"
    />
  </View>
</View>
`;
